src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,142 @@
+            ArrayList<String> tagList = new ArrayList<>();
+            String[] tagToken = tagString.split("#");
+            if (tagToken.length > 1) {
+                for (int i = 1; i < tagToken.length; i++) {
+                    tagList.add(tagToken[i]);
+                }
+            }
+            Todo temp = new Todo(token[2], tagList);
+            checkAndSetDone(token, temp);
+            return temp;
+        } else if (token[0].equals("D")) {
+            if (token.length == 4) {
+                Deadline temp = new Deadline(token[2], LocalDate.parse(token[3]));
+                checkAndSetDone(token, temp);
+                return temp;
+            }
+            String tagString = token[4];
+            ArrayList<String> tagList = new ArrayList<>();
+            String[] tagToken = tagString.split("#");
+            if (tagToken.length > 1) {
+                for (int i = 1; i < tagToken.length; i++) {
+                    tagList.add(tagToken[i]);
+                }
+            }
+            Deadline temp = new Deadline(token[2], tagList, LocalDate.parse(token[3]));
+            checkAndSetDone(token, temp);
+            return temp;
+        } else if (token[0].equals("E")) {
+            if (token.length == 5) {
+                Event temp = new Event(token[2], LocalDate.parse(token[3]), LocalDate.parse(token[4]));
+                checkAndSetDone(token, temp);
+                return temp;
+            }
+            String tagString = token[5];
+            ArrayList<String> tagList = new ArrayList<>();
+            String[] tagToken = tagString.split("#");
+            if (tagToken.length > 1) {
+                for (int i = 1; i < tagToken.length; i++) {
+                    tagList.add(tagToken[i]);
+                }
+            }
+            Event temp = new Event(token[2], tagList, LocalDate.parse(token[3]), LocalDate.parse(token[4]));
+            checkAndSetDone(token, temp);
+            return temp;
+        } else {
+            return null;
+        }
+    }
+
+    private void checkAndSetDone(String[] token, Task temp) {
+        if (token[1].equals("1")) {
+            temp.setDone();
+        }
+    }
+
+    private String writeTask(Task t) {
+        String taskLine = "";
+        String icon = t.getTaskTypeIcon();
+        int status = t.getStatus() ? 1 : 0;
+        String description = t.getDescription();
+        String tags = t.printTags();
+        if (icon.equals("T")) {
+            taskLine = String.format("%s,%s,%s,%s\n", icon, status, description, tags);
+        } else if (icon.equals("D")) {
+            taskLine = String.format("%s,%s,%s,%s,%s\n", icon, status, description, ((Deadline) t).getBy(), tags);
+        } else if (icon.equals("E")) {
+            taskLine = String.format("%s,%s,%s,%s,%s,%s\n",
+                    icon, status, description, ((Event) t).getFrom(), ((Event) t).getTo(), tags);
+        }
+        return taskLine;
+    }
+
+    /**
+     * Load the file into a list.
```

Comment:
Should be Loads rather than Load. The same applies to the remaining Javadoc comments with similar structure.